others/iterative_tree_traversals.cpp

```diff
@@ -0,0 +1,187 @@
+/**
+ * @file
+ * @brief Iterative version of Preorder, Postorder and preorder traversal of tree,
+ (https://en.wikipedia.org/wiki/Tree_traversal).
+ * @author [Motasim](https://github.com/motasimmakki)
+ * @details
+ * 
+ * ### Iterative Preorder Traversal of a tree
+ * Create a Stack that will store the Node of Tree.
+ * Push the root node into the stack.
+ * Save the root into the variabe named as current, and pop and elemnt from the stack.
+ * Store the data of current into the result array, and start traversing from it.
+ * Push both the child node of the current node into the stack, first right child then left child.
+ * Repeat the same set of steps untill the Stack becomes empty.
+ * And return the result array as the preorder traversal of a tree.
+ * 
+ * ### Iterative Postorder Traversal of a tree
+ * Create a Stack that will store the Node of Tree.
+ * Push the root node into the stack.
+ * Save the root into the variabe named as current, and pop and elemnt from the stack.
+ * Store the data of current into the result array, and start traversing from it.
+ * Push both the child node of the current node into the stack, first left child then right child.
+ * Repeat the same set of steps untill the Stack becomes empty.
+ * Now reverse the result array and then return it to the calling function as a postorder traversal of a tree.
+ * 
+ * ### Iterative Inorder Traversal of a tree
+ * Create a Stack that will store the Node of Tree.
+ * Push the root node into the stack.
+ * Save the root into the variabe named as current.
+ * Now iterate and take the current to the extreme left of the tree by traversing only to its left.
+ * Pop the elemnt from the stack and assign it to the current.
+ * Store the data of current into the result array.
+ * Repeat the same set of steps until the Stack becomes empty or the current becomes NULL.
+ * And return the result array as the inorder traversal of a tree.
+ */
+#include <iostream>   /// for I/O operations
+#include <cstdlib>    /// for `malloc`
+#include <stack>      /// for `stack`
+#include <vector>     /// for `vector`
+#include <algorithm>  /// for `reverse`
+
+// struct that defines the structure of the node of a tree.
+using Node = struct BinaryTree{
+    int data;
+    struct BinaryTree *left;
+    struct BinaryTree *right;
+};

```

Comment:
Also there are functions which are associated with this `struct`, then why don't you use a `class`?